Modules/_testcapimodule.c

```diff
@@ -1559,6 +1559,19 @@ getargs_et_hash(PyObject *self, PyObject *args)
     return result;
 }
 
+static PyObject *
+get_indices(PyObject *self, PyObject *args)
+{
+  PySliceObject *slice;
+  Py_ssize_t length, start, stop, step;
+
+  if (!PyArg_ParseTuple(args, "On", &slice, &length))
+    return NULL;
+
+  int result = PySlice_GetIndices(slice, length, &start, &stop, &step);
+  return Py_BuildValue("i", result);
```

Comment:
Is it worth to return also start, stop and step?

Reply:
I tried this but since when `r->start` is a float then -1 is returned and *start is not set. *start points to value like 140737488324304 that was there during initialization which I can't compare in the test case. Any thoughts on this?

Reply:
What if return a tuple only if result != -1?
```c
if (result == -1) {
    PY_RETURN_NONE;
}
return Py_BuildValue("innn", result, r->start, r->stop, r->step);
```